### PR TITLE
optimize:Use the openjdk image of eclipse-temurin as the base image

### LINKED ISSUE #6916: Replace base image with temurin
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

Minimal yet complete reproducer code (or URL to code):

### Ⅵ. Environment:

- JDK version(e.g. `java -version`):
- Seata client/server version:  
- Database version:
- OS(e.g. `uname -a`):
- Others:

### PR DESCRIPTION
**Pull request by xingfudeshi**
- [x] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did
Use the openjdk image of eclipse-temurin as the base image

### Ⅱ. Does this pull request fix one issue?

fixes #6916

### Ⅳ. Describe how to verify it
image
image

### CODE REVIEW

File: .github/workflows/publish-docker.yml [hunk 1]
```diff
@@ -39,10 +39,9 @@ jobs:
           REGISTRY_PASSWORD: ${{ secrets.DOCKERHUB_TOKEN }}
         run: |
           if [ "${{ github.ref_name }}" == "develop" ] || [ "${{ github.ref_name }}" == "snapshot"  || [ "${{ github.ref_name }}" == "2.x" ]; then
-            ./mvnw -T 4C clean package -Dimage.name=openjdk:8u342      -Pimage                                    -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
+            ./mvnw -T 4C clean package -Dimage.name=eclipse-temurin:8-jdk      -Pimage                                    -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
```
Comment (funky-eyes) [inline .github/workflows/publish-docker.yml]:
```suggestion
            ./mvnw -T 4C clean package -Dimage.name=eclipse-temurin:8u422-b05-jdk      -Pimage                                    -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
```
为什么不指定版本？
Why don't you specify the version?

File: .github/workflows/publish-docker.yml [hunk 2]
```diff
@@ -39,10 +39,9 @@ jobs:
...
           else
-            ./mvnw -T 4C clean package -Dimage.name=openjdk:8u342      -Pimage,release-image-based-on-java8       -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
-            ./mvnw -T 4C clean package -Dimage.name=openjdk:8u342-slim -Pimage,release-image-based-on-java8-slim  -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
+            ./mvnw -T 4C clean package -Dimage.name=eclipse-temurin:8-jdk      -Pimage,release-image-based-on-java8       -DskipTests -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
```
Comment (funky-eyes) [inline .github/workflows/publish-docker.yml]: ditto

Comment (xingfudeshi) [reply .github/workflows/publish-docker.yml]: Done.

**Review by wangliang181230 — APPROVED** [review]
LGTM

**Review by funky-eyes — APPROVED** [review]
LGTM

**Pull request merged**